network/network.go

```diff
@@ -1331,3 +1377,75 @@ func (n *network) restartOnDisconnect() {
 		}
 	}
 }
+
+// Health returns information about several network layer health checks.
+// 1) Information about health check results
+// 2) An error if the health check reports unhealthy
+// Assumes [n.stateLock] is not held
+func (n *network) Health() (interface{}, error) {
+	details := map[string]interface{}{}
+	healthy := true
+
+	// Get some data with the state lock held
+	connectedTo := 0
+	n.stateLock.RLock()
+	for _, peer := range n.peers {
+		if peer != nil && peer.connected.GetValue() {
+			connectedTo++
+			if connectedTo > int(n.healthConfig.MinConnectedPeers) {
+				break
+			}
+		}
+	}
+	pendingSendBytes := n.pendingBytes
+	numSendFailed := n.sendFailMeter.Ticks()
+	numSendSuccess := n.sendSuccessMeter.Ticks()
+	n.stateLock.RUnlock()
+
+	// Make sure we're connected to at least the minimum number of peers
+	if connectedTo < int(n.healthConfig.MinConnectedPeers) {
+		healthy = false
+		details["connectedToMinPeers"] = false
+	} else {
+		details["connectedToMinPeers"] = true
+	}
+
+	// Make sure we've received an incoming message within the threshold
+	lastMsgReceivedAt := time.Unix(atomic.LoadInt64(&n.lastMsgReceivedTime), 0)
+	timeSinceLastMsgReceived := n.clock.Time().Sub(lastMsgReceivedAt)
+	if timeSinceLastMsgReceived > n.healthConfig.MaxTimeSinceMsgReceived {
+		healthy = false
+	}
```

Comment:
I feel like I'd prefer no `if` here as well (and below), but let me know what you think